splunklogger.js

```diff
@@ -169,16 +250,66 @@ SplunkLogger.prototype._initializeConfig = function(config) {
         ret.protocol = config.protocol || ret.protocol || defaultConfig.protocol;
         ret.level = config.level || ret.level || defaultConfig.level;
 
+        ret.maxRetries = config.maxRetries || ret.maxRetries || defaultConfig.maxRetries;
+        ret.maxRetries = parseInt(ret.maxRetries, 10);
+        if (isNaN(ret.maxRetries)) {
+            throw new Error("Max retries must be a number, found: " + ret.maxRetries);
+        }
+        else if (ret.maxRetries < 0) {
+            throw new Error("Max retries must be a positive number, found: " + ret.maxRetries);
+        }
+
         // Start with the default autoFlush value
         ret.autoFlush = defaultConfig.autoFlush;
         // Then check this.config.autoFlush
         if (this.hasOwnProperty("config") && this.config.hasOwnProperty("autoFlush")) {
-            ret.autoFlush = ret.autoFlush;
+            ret.autoFlush = this.config.autoFlush;
         }
         // Then check the config.autoFlush, the function argument
         if (config.hasOwnProperty("autoFlush")) {
             ret.autoFlush = config.autoFlush;
         }
+        // Convert autoFlush value to boolean
+        ret.autoFlush = !!ret.autoFlush;
+
+        ret.maxBatchCount = config.maxBatchCount || ret.maxBatchCount || defaultConfig.maxBatchCount;
+        ret.maxBatchCount = parseInt(ret.maxBatchCount, 10);
+        if (isNaN(ret.maxBatchCount)) {
+            throw new Error("Max batch count must be a number, found: " + ret.maxBatchCount);
+        }
+        else if (ret.maxBatchCount < 0) {
+            throw new Error("Max batch count must be a positive number, found: " + ret.maxBatchCount);
+        }
+
+        ret.maxBatchSize = config.maxBatchSize || ret.maxBatchSize || defaultConfig.maxBatchSize;
+        ret.maxBatchSize = parseInt(ret.maxBatchSize, 10);
+        if (isNaN(ret.maxBatchSize)) {
+            throw new Error("Max batch size must be a number, found: " + ret.maxBatchSize);
+        }
+        else if (ret.maxBatchSize < 0) {
+            throw new Error("Max batch size must be a positive number, found: " + ret.maxBatchSize);
+        }
+
+        ret.batchInterval = config.batchInterval || ret.batchInterval || defaultConfig.batchInterval;
+        ret.batchInterval = parseInt(ret.batchInterval, 10);
+        if (isNaN(ret.batchInterval)) {
+            throw new Error("Batch interval must be a number, found: " + ret.batchInterval);
+        }
+        else if (ret.batchInterval < 0) {
+            throw new Error("Batch interval must be a positive number, found: " + ret.batchInterval);
+        }
+
+        var startTimer = !this._timerID && ret.autoFlush && ret.batchInterval > 0;
```

Comment:
These are a bit too "cute" - you at least need to add a comment explaining the logic.